sycl/test-e2e/OneapiDeviceSelector/illegal_input.cpp

```diff
@@ -1,15 +1,15 @@
 
 // RUN: %clangxx -fsycl -fsycl-targets=%{sycl_triple} %S/Inputs/trivial.cpp -o %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="macaroni:*" %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR=":" %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="level_zero:." %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="macaroni_level_zero:." %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="level_zero:macaroni_gpu" %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="level_zero:0..0" %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="level_zero:" %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="level_zero:::gpu" %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="level_zero:.1" %{run-unfiltered-devices} %t.out
-// RUN: not env ONEAPI_DEVICE_SELECTOR="" %{run-unfiltered-devices} %t.out
+// RUN: not --crash env ONEAPI_DEVICE_SELECTOR="macaroni:*" %{run-unfiltered-devices} %t.out
```

Comment:
sorry what does `--crash` do?

Reply:
The workaround for the lit hang that I was experiencing involved calling the `_runShTest` function with the `useExternalSh` parameter set to true (previously this was false). When lit uses the internal shell it has some special logic for handling the `not` command which I think made it unnecessary to add the `--crash` flag here. However this failed after adding the workaround because the underlying executable that was called crashed instead of exiting normally with a non-zero exit code, and the `not` command in the external shell needed this flag to properly react to that crash.

I also ran this test using the internal shell + the `--crash` flag and it works fine too.

Reply:
ah ok thanks for the explanation, i thought it was running bash `not` here and i couldn't find any info about that arg, but we're actually calling an llvm utility also named `not`, where the `--crash` argument makes it return true if the thing crashes. the more you know.